### PR TITLE
fix(test): test case for snap prep

### PR DESCRIPTION
**Pull request by pawanpraka1**
adding test cases for SNAP_PREP ops.

Signed-off-by: Pawan <pawan@mayadata.io>

### CODE REVIEW

**Review by mynktl — APPROVED** [review]

File: src/mock_client.c [hunk 1]
```diff
@@ -292,9 +292,18 @@ snapshot_thread(void *args)
 		io_wait_time = random() % 2 + 2;
 		wait_time = random() % 2 + 4;
 
+		MTX_LOCK(&spec->rq_mtx);
```
Comment (vishnuitta) [inline src/mock_client.c]: comment would be good to understand why we are doing this when we revisit this code

**Review by vishnuitta — APPROVED** [review]
changes are good.. lets handle timeout case as well in new PR

**Comment by vishnuitta** [comment]
@pawanpraka1 can you look at travis failure

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**